src/code/z_kaleido_setup.c

```diff
@@ -1,9 +1,145 @@
 #include "global.h"
+#include "overlays/gamestates/ovl_file_choose/z_file_choose.h"
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_setup/func_800F4A10.s")
+s16 D_801BDB00[] = { PAUSE_1, PAUSE_2, PAUSE_3, PAUSE_0 };
+f32 sKaleidoSetupEyeX[] = { -64.0f, 0.0f, 64.0f, 0.0f };
+f32 sKaleidoSetupEyeZ[] = { 0.0f, -64.0f, 0.0f, 64.0f };
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_setup/func_800F4C0C.s")
+void func_800F4A10(GlobalContext* globalCtx) {
+    PauseContext* pauseCtx = &globalCtx->pauseCtx;
+    s16 i;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_setup/KaleidoSetup_Init.s")
+    func_8013EE24();
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_kaleido_setup/KaleidoSetup_Destroy.s")
+    pauseCtx->unk_206 = 0;
+    pauseCtx->unk_200 = 1;
+    pauseCtx->eye.x = sKaleidoSetupEyeX[pauseCtx->pageIndex];
+    pauseCtx->eye.z = sKaleidoSetupEyeZ[pauseCtx->pageIndex];
+    pauseCtx->pageIndex = D_801BDB00[pauseCtx->pageIndex];
+    pauseCtx->unk_27E = -40;
+
+    for (i = 0; i < ARRAY_COUNT(pauseCtx->worldMapPoints); i++) {
+        pauseCtx->worldMapPoints[i] = 0;
+    }
+
+    if (pauseCtx->state == 1) {
+        for (i = 0; i < 11; i++) {
+            if ((gSaveContext.save.mapsVisited >> i) & 1) {
+                pauseCtx->worldMapPoints[i] = 1;
+            }
+        }
+    } else {
+        for (i = 9; i >= 0; i--) {
+            if ((gSaveContext.save.playerData.owlActivationFlags >> i) & 1) {
+                pauseCtx->worldMapPoints[i] = 1;
+                pauseCtx->unk_238[4] = i;
+            }
+        }
+
+        if ((gSaveContext.save.playerData.owlActivationFlags >> 4) & 1) {
+            pauseCtx->unk_238[4] = 4;
+        }
+    }
```

Comment:
I suppose the indices here could get defines or something.

Reply:
You want a define for `4`?

Reply:
Well, the for loop ones.

Reply:
Well, yes, it would be nice to have an enum for those values, but I don't know what those are, and those even aren't used explicitly on this code snippet. So I think this enum should be made by whoever decomps  kaleido in the future